script/data-harmonizer/index.js

```diff
@@ -1011,6 +1011,15 @@ let DataHarmonizer = {
 				new_field['title'] = new_field['name'];
 			}
 
+			// The presence of a min/max value implies that this field is numeric
```

Comment:
@pkalita-lbl can you provide an example of a slot that asserts a min and/or max but not a range?

Reply:
@ddooley you're right that this absolutely is not a good long-term solution. My goal was do something minimally invasive for now to get min/max validation working again for the next NMDC release. 

@turbomam `dna_concentration`, `dna_volume` are two that have min/max but no `range`. `ph` has min/max values, but also has `range: string` which doesn't really make too much sense

Reply:
That's probably not an exhaustive list of slots! Those were just the first few I noticed when search through `nmdc_dh.yml`

Reply:
The pH silliness is coming from [here](https://docs.google.com/spreadsheets/d/1cMlPKgjZh-v21aMYCm9x1TxzE5BwGQptBQcvuaYAtC8/edit#gid=1241218612&range=46:46). I think I forced the range to be a string so that regular expression checking would be applied.

@ddooley or @pkalita-lbl can you show me if/where/how DH checks LinkML ranges and applies (xsd?) type-based validation? If that's in place in this branch, when I will definitely remove that bogus string range assertion on pH

Reply:
When a template is loaded, the `range` slot is translated into a new `datatype` field [here](https://github.com/cidgoh/DataHarmonizer/blob/linkml-datastructure/script/data-harmonizer/index.js#L1015). When validation happens, that `datatype` field is checked [here](https://github.com/cidgoh/DataHarmonizer/blob/linkml-datastructure/script/data-harmonizer/validation.js#L58) and various type-specific checks are done.

Reply:
About min and max, I think we should detect given data type - numeric or date, and base range check on that.  I recall master branch behaves that way so I'll ensure linkml datastructure does too.  But that means onus is on linkml schema to define datatype for any min max value items.

Also I'm making it a priority this week to ensure ranges that include more than one enumeration source are implemented if DH linkml datastructure doesn't already do that.

Reply:
> I think we should detect given data type - numeric or date, and base range check on that. I recall master branch behaves that way so I'll ensure linkml datastructure does too.

I believe the `linkml-datastructure` branch does that already.

> But that means onus is on linkml schema to define datatype for any min max value slots

Yes, exactly. My change here is essentially a workaround to _not_ have that onus on the schema. But having it dictated by the schema is the right move. Since @turbomam is working on the schema, I'm happy to remove this controversial bit from this branch and just leave the guidance stuff.